webapp/app/components/pages/login/LoginPage.test.js

```diff
@@ -14,63 +12,6 @@ describe('<LoginPage /> component', () => {
       actions: {}
     });
 
-    expect(wrapper.find('section')).toHaveLength(1);  
-    expect(wrapper.find('LoginForm')).toHaveLength(1);
-  });
-
-  it('should handle form submit itself', () => {
-    const login = jest.fn();
-    const wrapper = setup({
-      actions: {
-        login
-      }
-    });
-    const form = wrapper.find('LoginForm');
-
-    form.simulate('submit');
-
-    expect(login).toHaveBeenCalledTimes(1);
-  });
-
-  describe('mapStateToProps functions', () => {
-    it('should return the initial state of auth module', () => {
-      const expectedProps = {
-        authenticating: false,
-        isAuthenticated: false,
-        error: false,
-        errorMessage: null,
-        user: null
-      };
-
-      const props = mapStateToProps(initialState);
-
-      expect(props).toEqual(expectedProps);
-    });
-  });
-
-  describe('mapDispatchToProps functions', () => {
-    it('actions prop should be defined', () => {
-
-      const dispatch = () => {};
-
-      const props = mapDispatchToProps(dispatch);
-
-      expect(props.actions).toBeDefined();
-    });
-
-    it('should return the binded actions', () => {
-
-      const dispatch = () => {};
-      const expectedActions = [
-        'loginRequest',
-        'loginSuccess',
-        'loginFailed',
-        'login'
-      ];
-
-      const props = mapDispatchToProps(dispatch);
-
-      expect(keys(props.actions)).toEqual(expectedActions);
-    });
+    expect(wrapper.find('section')).toHaveLength(1);
```

Comment:
Please add missing tests to make this one and all other new tests valid and more robust,  

Testing we have a section element is valid, but alongside all other components and elements we are expecting to have in the rendered component.

otherwise testing we just have a section `expect(wrapper.find('section')).toHaveLength(1);`
is fragile since the section element could be completely empty and the test will still pass.

We should test against functionality rather than implementation.

The more robust way to test here would be to check against the components we are expecting and the actual output rendered.

There at least 5 expectations missing here.

PS: Same applies to all other tests

LoginNavigation
```
it('should have one <LoginNavigation>', () => {
    expect(wrapper.find('LoginNavigation').length).toBe(1);
});
```

Container
```
it('should have one `div` with class `container`', () => {
    expect(wrapper.find('div.container').length).toBe(1);
});
```

Card
```
it('should have one card `div` with class `card-container`', () => {
    expect(wrapper.find('div.card.card-container').length).toBe(1);
});
```

Profile Image
```
it('should have one `image` with class `profile-img`', () => {
    // TODO additionaly expectation on the url as well
    expect(wrapper.find('img.profile-img').length).toBe(1);
});
```

LoginFormMain
```
it('should have one `<LoginFormMain>` ', () => {
    expect(wrapper.find('LoginFormMain').length).toBe(1);
});
```

Reply:
I put a comment in the initial PR when was created:
Tests and clean code are pending

I'm working on this, base PR.